dev/com.ibm.ws.org.jboss.weld/src/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -0,0 +1,1001 @@
+import static org.jboss.classfilewriter.util.DescriptorUtils.isWide;
+import static org.jboss.weld.util.reflection.Reflections.cast;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.Bean;
+
+import org.jboss.classfilewriter.AccessFlag;
+import org.jboss.classfilewriter.ClassFile;
+import org.jboss.classfilewriter.ClassMethod;
+import org.jboss.classfilewriter.DuplicateMemberException;
+import org.jboss.classfilewriter.code.BranchEnd;
+import org.jboss.classfilewriter.code.CodeAttribute;
+import org.jboss.classfilewriter.util.Boxing;
+import org.jboss.classfilewriter.util.DescriptorUtils;
+import org.jboss.weld.Container;
+import org.jboss.weld.bean.builtin.AbstractBuiltInBean;
+import org.jboss.weld.config.WeldConfiguration;
+import org.jboss.weld.exceptions.DefinitionException;
+import org.jboss.weld.exceptions.WeldException;
+import org.jboss.weld.interceptor.proxy.LifecycleMixin;
+import org.jboss.weld.interceptor.util.proxy.TargetInstanceProxy;
+import org.jboss.weld.logging.BeanLogger;
+import org.jboss.weld.security.GetDeclaredConstructorsAction;
+import org.jboss.weld.security.GetDeclaredMethodsAction;
+import org.jboss.weld.security.GetProtectionDomainAction;
+import org.jboss.weld.serialization.spi.BeanIdentifier;
+import org.jboss.weld.serialization.spi.ContextualStore;
+import org.jboss.weld.serialization.spi.ProxyServices;
+import org.jboss.weld.util.Proxies;
+import org.jboss.weld.util.Proxies.TypeInfo;
+import org.jboss.weld.util.bytecode.BytecodeUtils;
+import org.jboss.weld.util.bytecode.ClassFileUtils;
+import org.jboss.weld.util.bytecode.ConstructorUtils;
+import org.jboss.weld.util.bytecode.DeferredBytecode;
+import org.jboss.weld.util.bytecode.MethodInformation;
+import org.jboss.weld.util.bytecode.RuntimeMethodInformation;
+import org.jboss.weld.util.collections.Arrays2;
+import org.jboss.weld.util.collections.ImmutableSet;
+import org.jboss.weld.util.collections.Sets;
+import org.jboss.weld.util.reflection.Reflections;
+
+
+/**
+ * Main factory to produce proxy classes and instances for Weld beans. This
+ * implementation creates proxies which forward non-static method invocations to
+ * a {@link BeanInstance}. All proxies implement the {@link Proxy} interface.
+ *
+ * @author David Allen
+ * @author Stuart Douglas
+ * @author Marius Bogoevici
+ * @author Ales Justin
+ */
+public class ProxyFactory<T> implements PrivilegedAction<T> {
```

Comment:
As a requirement of the Apache 2.0 license (4b), this file needs a prominent notice saying that we have modified it.

In addition, we should outline how and why we modified it in the class doc and add comments around all of our changes.